src/Types.hpp

```diff
@@ -0,0 +1,13 @@
+#ifndef _TYPES_HPP_
```

Comment:
If we're moving generic types here, lets also move `LocalDate` and `PointOfInterest`

Reply:
Separate PR for that.

Reply:
Fair